### PR TITLE
JDK-8315486: vmTestbase/nsk/jdwp/ThreadReference/ForceEarlyReturn/forceEarlyReturn002/forceEarlyReturn002.java timed out

### PR DESCRIPTION
**Pull request by alexmenkov**
To test ForceEarlyReturn command for NO_MORE_FRAMES case the test creates ThreadStartEventRequest with SUSPEND_ALL policy and requests debuggee to start new thread.
If debuggee JVM starts some internal threads before the request is cleared (i.e. we have several ThreadStart events), 2nd event suspends debuggee again and the test fails with timeout.
The change adds THREAD_ONLY modifier to the ThreadStartEventRequest to generate event only for desired thread.
It requires thread ID, so debuggee was updated to create Thread object in advance, debugger reads the thread ID from static field (it does not need to be static, but Debugee class has convenient methods to retrieve class ID and static field value).

Testing: 100 runs of the test on windows-x64-debug,linux-x64-debug,macosx-x64-debug with JTREG_TEST_THREAD_FACTORY=Virtual, with and without "-XX:+UseZGC -XX:+ZGenerational"

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315486](https://bugs.openjdk.org/browse/JDK-8315486): vmTestbase/nsk/jdwp/ThreadReference/ForceEarlyReturn/forceEarlyReturn002/forceEarlyReturn002.java timed out (**Bug** - P4)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [310115de](https://git.openjdk.org/jdk/pull/15601/files/310115de84675d505c90242f9bac756b3a56f21b)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15601/head:pull/15601` \
`$ git checkout pull/15601`

Update a local copy of the PR: \
`$ git checkout pull/15601` \
`$ git pull https://git.openjdk.org/jdk.git pull/15601/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15601`

View PR using the GUI difftool: \
`$ git pr show -t 15601`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15601.diff">https://git.openjdk.org/jdk/pull/15601.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15601#issuecomment-1709268298)

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/ForceEarlyReturn/forceEarlyReturn002/forceEarlyReturn002.java [hunk 1]
```diff
@@ -139,13 +139,32 @@ private void sendCommand(long threadID, Value value, boolean expectError, int er
...
+
     private int createThreadStartEventRequest() {
         try {
+            long newThreadId = getNewThreadId();
             // create command packet and fill requred out data
             CommandPacket command = new CommandPacket(JDWP.Command.EventRequest.Set);
             command.addByte(JDWP.EventKind.THREAD_START);
             command.addByte(JDWP.SuspendPolicy.ALL);
-            command.addInt(0);
+            // THREAD_ONLY modifier
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/ForceEarlyReturn/forceEarlyReturn002/forceEarlyReturn002.java:164]: Maybe add a bit more of a comment here that explains we only want the THREAD_START event for the specified test thread and not any others that might start up.

**Review by plummercj — APPROVED** [review]
Overall looks good. Just one minor comment suggestion.

**Comment by alexmenkov** [comment]
Ping. need 1 more review.

**Review by lmesnik — APPROVED** [review]

**Pull request closed without merging**